### PR TITLE
Add static builds for various platforms

### PR DESCRIPTION
**Pull request by jagerman**
Adds static bundles (that get uploaded via CI) for various platforms:
- Linux amd64/i386/arm64/armhf
- Windows x64/i686
- macOS (single multiarch lib for both x86_64 and arm64)
- Android (x86_64, armeabi-v7a, arm64-v8a, and x86)
- iOS (arm64 for devices + multiarch lib for x86_64 arm64 simulators)

### CODE REVIEW

**Comment by majestrate** [comment]
i like the way things that dont belong in contrib arent in contrib and want to add that to lokinet.

File: utils/ios.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+set -e
+
+if ! [ -f LICENSE ] || ! [ -d include/session ]; then
+    echo "You need to run this as ./utils/ios.sh from the top-level libsession-util project directory"
```
Comment (majestrate) [inline utils/ios.sh]: should this write to stderr as the one below does ?

File: utils/ios.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+set -x
+
+for i in "ios-arm64 OS64" "sim-arm64 SIMULATORARM64" "sim-x86_64 SIMULATOR64"; do
+    i=($i)
+    build="build-ios/${i[0]}"
+    platform="${i[1]}"
+    echo "Building ${i[0]} for $platform in $build"
+
+    ./utils/static-bundle.sh "$build" "" \
+        -DCMAKE_TOOLCHAIN_FILE=../../external/ios-cmake/ios.toolchain.cmake \
```
Comment (majestrate) [inline utils/ios.sh:39]: use absolute path here as having it be a relative the way it is hard to understand.

File: utils/ios.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+
+# Create a '.xcframework' so XCode can deal with the different architectures
+xcodebuild -create-xcframework \
+    -library build-ios/ios-arm64/libsession-util.a \
+    -library build-ios/sim/libsession-util.a \
+    -output "$pkg_dir/libsession-util.xcframework"
+
+# Copy the headers over
+cp -rv include/session $pkg_dir/libsession-util.xcframework
```
Comment (majestrate) [inline utils/ios.sh]: should those two paths to the xcframework be escaped the same?

File: utils/ios.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+# The 'module.modulemap' is needed for XCode to be able to find the headers
+modmap="$pkg_dir/libsession-util.xcframework/module.modulemap"
```
Comment (majestrate) [inline utils/ios.sh:63]: i feel like cmake should make this file honestly

File: utils/ios.sh (hunk 2 above)
Comment (jagerman) [reply utils/ios.sh:39]: Relative is easier when you are running the same commands manually while looking at the script (e.g. to debug something).

File: utils/ios.sh (hunk 4 above)
Comment (jagerman) [reply utils/ios.sh:63]: This is only something ios cares about so not worth the bother.

**Pull request merged**